client/src/modules/stock/entry/modals/lots.modal.js

```diff
@@ -309,12 +325,26 @@ function StockDefineLotsModalController(
     });
 
     if (vm.errors.length === 0) {
-      saveSetting();
-      Instance.close({
-        lots : vm.form.rows,
-        unit_cost : vm.stockLine.unit_cost,
-        quantity : vm.form.total(),
-      });
+
+      // Maybe update some lot expiration dates
+      const promises = [];
+      if (vm.editExpirationDates) {
+        vm.form.rows.forEach((row) => {
+          const existingLot = getExistingLot(row.lot);
+          if (existingLot && (row.expiration_date !== existingLot.expiration_date)) {
+            promises.push(Lots.update(existingLot.uuid, { expiration_date : row.expiration_date }));
+          }
+        });
+      }
+      Promise.all(promises)
```

Comment:
We need to ensure that, if this delays the modal (imagine editing 1000 lots over a slow network), we have a loading indicator. 

Could you transition the `lots.modal.html` file to using `bh-submit` with a `bh-loading-button`?  Then you can just return this Promise and you'll get a loading indicator for free.

Reply:
@jniles Please take a look at my fix for this.  If it okay and you think it is ready to release, let me know and I'll rebase and release.

Reply:
You need to return this `Promise.all()` for it to have the intended effect.  The magic happens because the `submit()` method must return a promise.